### DISCUSSION TITLE
Tokio vs async_std

### DISCUSSION THREAD

**Discussion by tjdevries** [tjdevries/rofl.nvim#5] [General]
Why would you use tokio over async_std? Do you have anything I can read about for them?

**Comment by oberblastmeister** [comment]
I use tokio over async_std because it is more mature. For tokio there are the [docs](https://docs.rs/tokio/1.1.1/tokio/) and the [website](https://tokio.rs/) which has a tutorial.

**Reply by oberblastmeister** [reply]
Some opinions on the rust reddit:
https://www.reddit.com/r/rust/comments/dngig6/tokio_vs_asyncstd/f5ax3ub?utm_source=share&utm_medium=web2x&context=3
https://www.reddit.com/r/rust/comments/dngig6/tokio_vs_asyncstd/f5b4oao?utm_source=share&utm_medium=web2x&context=3

**Reply by tjdevries** [reply]
TY for the reading :)

(If reddit likes tokio better, then it must be the best :laughing: )

**Reply by oberblastmeister** [reply]
yes reddit rules. In general the async ecosytem is still very new so thats why there are hacks like `async_trait`

**Reply by tjdevries** [reply]
:+1: OK, that's good to know. I'm fine with either. On chat, people recommend `async_std` since it seemed to be gaining more traction, but I don't really care either way.

If you submit PR to switch to tokio, I will merge it.

**Reply by lucastrvsn** [reply]
Tokio is been heavily used on [deno](https://github.com/denoland/deno) project as well. My vote is for tokio!

**Reply by OldManFroggy** [reply]
def. tokio v1+, async_std is getting more popular but majority use tokio right now.  The interface for tokio and async_std are similar so could add the other as optional feature down the line if ever needed.   

plus. more people know and understand tokio and its issues, and how to work around them,
